frontend/src/components/ThemedInput/ThemedTextArea.tsx

```diff
@@ -92,11 +96,23 @@ function ThemedTextArea({
 
 	function inputChange(event: React.ChangeEvent<HTMLTextAreaElement>) {
 		onContentChanged(event.target.value);
+		if (validateInput) {
+			validateInput(event.target.value);
+		}
 	}
 
-	const textAreaClass = clsx('themed-input', 'themed-text-area', spacing, {
-		disabled,
-	});
+	// check config validated exists and is not false
+	const validInput = configValidated === false ? 'invalid-input' : '';
+
+	const textAreaClass = clsx(
+		'themed-input',
+		'themed-text-area',
+		spacing,
+		validInput,
```

Comment:
clsx library is quite versatile; you can use this instead of needing to declare const validInput:

```
const textAreaClass = clsx(
  'themed-input',
  'themed-text-area',
  spacing,
  configValidated === false && 'invalid-input',
  { disabled }
);
```

Same idea in ThemedNumberInput.

Reply:
Alternatively, you could "reverse polarity" and have the prop (and the state in the parent components) be "isInvalid".

It makes me a touch uncomfortable that you need to check that the value is specifically `false` for an optional prop, because if you don't pass in a prop value it'll be undefined, which is "falsy". Better to invert the prop so "falsy" and "undefined" can be treated the same.

Does that make sense?